### DISCUSSION TITLE
Integrating Next.js within a PHP Page: How to Append Next.js Components to Specific DOM Elements?

### DISCUSSION THREAD

**Discussion by alihesari** [vercel/next.js#58537] [Help]
### Summary

I'm working on integrating Next.js (v14.0.2) into an existing PHP-based web application (Phalcon framework). I am not sure if I can incorporate Next.js components into specific sections of the PHP-generated page. I want to render React components generated by Next.js within targeted elements based on certain conditions or actions in the PHP logic.

I've attempted to use ReactDOM.createRoot in a manner similar to standalone React applications, but I'm aware that Next.js abstracts some of the rendering setup. How can I dynamically append Next.js-generated components to particular DOM elements within a PHP-rendered page? Are there specific Next.js methods or strategies to achieve this integration effectively without disrupting the PHP-rendered content? Any guidance, examples, or best practices would be highly appreciated. Thank you!

**Comment by olaleykhan** [comment]
@alihesari , Please were you able to get this to work?
